src/main/java/com/example/demo/validation/PasswordMatchesValidator.java

```diff
@@ -0,0 +1,15 @@
+package com.example.demo.validation;
+
+import com.example.demo.dto.user.UserRegistrationRequestDto;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.Objects;
+
+public class PasswordMatchesValidator
```

Comment:
For now, you can only use this annotation to compare passwords. It's better to make it independent so that it can be reused in other parts of the program.